### PR TITLE
fix(azure-ad): Add azure AD

### PR DESCRIPTION
**Pull request by PierrickV**
Migration from Logstash to Ingest of Azure AD

Associated issue https://github.com/SekoiaLab/platform/issues/24471

### CODE REVIEW

File: Azure/azure-ad/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  - name: set_if_message_properties
+    filter: '{{ parsed_event.message.properties != null }}'
+
+stages:
+  azuread:
+    actions:
+      - set:
+          service.type: "ldap"
+          service.name: "active directory"
+          log.hostname: "azureactivedirectory"
```
Comment (squioc) [inline Azure/azure-ad/ingest/parser.yml:19]: `host.hostname` is the standard field to hold the hostname
```suggestion
          log.hostname: "azureactivedirectory"
          host.hostname: "azureactivedirectory"
```

File: Azure/azure-ad/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          action.name: '{{parsed_event.message.operationName}}'
+          user.id: '{{ parsed_event.message.userId}}'
+          user.name: '{{ parsed_event.message.userPrincipalName}}'
+          action.outcome: '{{parsed_event.message.properties.result}}'
+          source.ip: '{{parsed_event.message.ipAddress}}'
+          source.address: '{{parsed_event.message.ipAddress}}'
```
Comment (squioc) [inline Azure/azure-ad/ingest/parser.yml:25]: Not required as The ingest worker set `source.address` from `source.ip` or `source.domain`
```suggestion
```

File: Azure/azure-ad/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        filter: "{{parsed_event.message.authenticationDetails != null}}"
+
+      - set:
+          source.ip: "{{parsed_event.message.callerIpAddress}}"
+        filter: "{{parsed_event.message.callerIpAddress != '<null>' and parsed_event.message.callerIpAddress != null }}"
+
+  set_if_message_properties:
+    actions:
+      - set:
+          action.outcome_reason: "{{parsed_event.message.properties.resultReason}}"
```
Comment (squioc) [inline Azure/azure-ad/ingest/parser.yml:46]: Add standard field:
```suggestion
          action.outcome_reason: "{{parsed_event.message.properties.resultReason}}"
          event.reason: "{{parsed_event.message.properties.resultReason}}"
```

File: Azure/azure-ad/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+          action.type: "{{parsed_event.message.properties.operationType |lower}}"
+
+      - set:
+          action.target: "user"
+        filter: "{{parsed_event.message.userId != null}}"
+
+  azuread-filters:
+    actions:
+      - set:
+          action.name: "authentication"
```
Comment (squioc) [inline Azure/azure-ad/ingest/parser.yml:36]: Add standard fields too:
```suggestion
          action.name: "authentication"
          event.category: "authentication"
          event.type: "start"
```

**Review by squioc — CHANGES_REQUESTED** [review]
Some standard fields to be compliant with the ECS specification

**Review by LeoArurault — DISMISSED** [review]
lgtm

File: Azure/azure-ad/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+          event.category: "authentication"
+          event.type: "start"
```
Comment (squioc) [inline Azure/azure-ad/ingest/parser.yml:38]: My bad. This field must be an array.
```suggestion
          event.type: ["start"]
```

**Review by squioc — CHANGES_REQUESTED** [review]

File: Azure/azure-ad/ingest/parser.yml (hunk 5 above)
Comment (PierrickV) [reply Azure/azure-ad/ingest/parser.yml]: Applied in https://github.com/SEKOIA-IO/intake-formats/pull/53/commits/1d1670493cf319f21c9a69fd4b2a32938bcc6b83

**Pull request merged**